pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1798,7 +1798,11 @@ func (l *LibvirtDomainManager) SignalShutdownVMI(vmi *v1.VirtualMachineInstance)
 	}
 
 	if domState == libvirt.DOMAIN_RUNNING || domState == libvirt.DOMAIN_PAUSED {
-		err = dom.ShutdownFlags(libvirt.DOMAIN_SHUTDOWN_ACPI_POWER_BTN)
+		flag := libvirt.DOMAIN_SHUTDOWN_DEFAULT
+		if vmi.Spec.Domain.Features != nil && vmi.Spec.Domain.Features.ACPI.Enabled != nil && *vmi.Spec.Domain.Features.ACPI.Enabled == true {
+			flag = libvirt.DOMAIN_SHUTDOWN_ACPI_POWER_BTN
+		}
+		err = dom.ShutdownFlags(flag)
```

Comment:
can you add a unit test to check that the right flags are being used?

Reply:
Done